### PR TITLE
[incubator/vault] Add support for prometheus operator

### PR DESCRIPTION
**Pull request by AlexDouze**
Signed-off-by: Alexandre Ledit <alexandre.ledit@easymile.com>

#### What this PR does / why we need it:

This PR add support for Prometheus Operator. It allows to create a ServiceMonitor resource for Prometheus to automatically scrape the vault-exporter. It also can create a PrometheusRules resource with some default rules, and custom ones.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
 - None

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/vault/templates/prometheusrules.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{{- if .Values.vaultExporter.alerts.enabled -}}
```
Comment (jbialy) [inline incubator/vault/templates/prometheusrules.yaml]: A small suggestion. How about leaving the `PrometheusRule` resource independent of `vaultExporter`. I'm thinking that the existing `vaultExporter` configuration might go away in the future in favour of something more generic (i.e. extraContainers). Could we go with something like this for enabling and defining the `PrometheusRule` resource:

```suggestion
{{- if .Values.prometheusRule.enabled -}}
```

This loosely following what other charts are doing:

https://github.com/helm/charts/blob/e5e3be5b957b3373bd233c03431189bd9c5ce7a7/stable/envoy/values.yaml#L254

File: incubator/vault/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{{- if .Values.vaultExporter.serviceMonitor.enabled -}}
```
Comment (jbialy) [inline incubator/vault/templates/servicemonitor.yaml]: Similarly to my above comment, let's have this configuration be independent of how `vaultExporter` is implemented:

```suggestion
{{- if .Values.serviceMonitor.enabled -}}
```

Another chart reference: https://github.com/helm/charts/blob/e5e3be5b957b3373bd233c03431189bd9c5ce7a7/stable/envoy/values.yaml#L243

**Comment by jbialy** [comment]
Hi @AlexDouze, thanks for the PR, this is great! I just wanted to point out a few suggestions, please see the comments.

**Comment by AlexDouze** [comment]
Hi @jbialy, I made the changes you suggested.
I had to add a new parameter `serviceMonitor.podPortName` in case you don't want to use the default vaultExporter and so a different port.
I added the related documentation in the README.

File: incubator/vault/README.md [hunk 3]
```diff
@@ -132,6 +143,60 @@ your needs.
...
+    ports:
+    - containerPort: 8080
+      name: metricPort
+```
+
+If you want to add Prometheus alerting rules you can simply enable the alerts and disabling/enabling
+the defaults rules you want to use. You can add as many custom rules as you want. For example:
+
+```yaml
+  alerts:
```
Comment (jbialy) [inline incubator/vault/README.md]: Should this be renamed? 

```suggestion
  prometheusRules:
```

**Comment by AlexDouze** [comment]
Thank you :)

**Pull request merged**